syntax = "proto3";
import "google/protobuf/any.proto";
package natashapb;

message SymbolLine {
    repeated int32 xDat = 1;
}

message SymbolBlock {
    repeated SymbolLine yDat = 1;
}

message SymbolBlock3X5 {
    int32 dat0_0 = 1;
    int32 dat0_1 = 2;
    int32 dat0_2 = 3;
    int32 dat0_3 = 4;
    int32 dat0_4 = 5;
    int32 dat1_0 = 6;
    int32 dat1_1 = 7;
    int32 dat1_2 = 8;
    int32 dat1_3 = 9;
    int32 dat1_4 = 10;
    int32 dat2_0 = 11;
    int32 dat2_1 = 12;
    int32 dat2_2 = 13;
    int32 dat2_3 = 14;
    int32 dat2_4 = 15;
}

message RandomResult {
    google.protobuf.Any info = 1;
}

message StaticCascadingRandomResult {
    int32 reelsIndex = 1;
    int32 downNums = 2;
}

message Position2D {
    int32 x = 1;
    int32 y = 2;
}

enum GAMERESULTTYPE {
    NULL = 0;
    LINE_LEFT = 1;
    LINE_RIGHT = 2;
    SCATTER_LEFT = 3;
    SCATTER_RIGHT = 4;
    SCATTER_LEFT_CONSEQUENT = 5;
    SCATTER_RIGHT_CONSEQUENT = 6;
};

message GameResultInfo {
    GAMERESULTTYPE typeGameResult = 1;
    repeated Position2D lstPos = 2;
    int32 symbol = 3;
    repeated int32 lstSymbol = 4;
    int64 win = 5;
    int64 realWin = 6;

    google.protobuf.Any exDat = 7;
}

message GameResultChunk {
    int64 win = 1;
    int64 realWin = 2;
    repeated GameResultInfo lstGRI = 3;

    google.protobuf.Any exDat = 4;
}